### ISSUE TITLE
Windows vs Linux differences in handling cred_acquire_cb

### ISSUE THREAD

**Issue by matthewsor** [libgit2/libgit2#3642]
I'm using the following version (using git describe): v0.16.0-7191-g9f4e7c8

On Linux if the user provided credentials are invalid (bad password) it will keep calling the cred_acquire_cb until it either returns an error or provides valid credentials (I think there is a limit of 7 times).  But on Windows this will only be called once due to logic differences in how the transports http.c and winhttp.c work. 

On windows in src/transports/winhttp.c around line 930 there is this logic:
			
```
			if (allowed_types &&
				(!t->cred || 0 == (t->cred->credtype & allowed_types)))
```
				
If this condition is true, the cred_acquire_cb is called
The first time through this works fine, the cred_acquire_cb is called and the user's credentials can be returned.
But if the user/pass is not correct and the credentials fail, the second time through here the !t->cred is false (but the credtype is one of the allowed) and as a result the callback is never called a second (or more times).

On Linux the corresponding function is in src/transports/http.c line 354
	
```
	/* Check for an authentication failure. */
	if (parser->status_code == 401 && get_verb == s->verb) {
		if (!t->owner->cred_acquire_cb) {
			no_callback = 1;
		} else {
			if (allowed_auth_types) {
				if (t->cred) {
					t->cred->free(t->cred);
					t->cred = NULL;
				}
```
				

It just clears out the t->cred and calls the callback without the extra logic about credtype and the extra requirement that there be no credentials.

As a result of this the Windows and Linux versions of the library behavior differ a whole lot on how they handle invalid credentials.

**Comment by carlosmn** [comment]
It looks like we never did activate the tests for the Windows CI, so we forgot that WinHTTP didn't get fixed like the others did. Theres #3648 to fix this.

**Comment by carlosmn** [comment]
Fixed via #3648
